templates/analysis.ng.yml

```diff
@@ -0,0 +1,101 @@
+# this template performs static analysis on an angular build project
+
+parameters:
+  - name: microservice  # <-- It is assumed that the angular portion will be the frontend
+    type: string        #     however, it can be overwritten should the need arise
+    default: frontend
+  - name: workDir       # <-- The working directory, can be relative or absolute, expects
+    type: string        #     a trailing '/' character.
+  - name: coveragePath  # <-- The path to all the files that should be included in the code
+    type: string        #     coverage
+
+  # external access parameters:
+  - name: sonarKeyPrefix
+    type: string
+  - name: sonarNamePrefix
+    type: string
+  - name: sonarOrg
+    type: string
+  - name: sonarUrlPrefix
+    type: string
+  - name: discordChannel
+    type: string
+  - name: discordKey
+    type: string
+
+steps:
+
+###  Need to come back and review this task later
+- task: SonarCloudPrepare@1
+  inputs:
+    SonarCloud: 'SonarCloud'
+    organization: ${{ parameters.sonarOrg }}
+    scannerMode: 'CLI'
+    configMode: 'manual'
+    cliProjectKey: ${{ parameters.sonarKeyPrefix }}${{ parameters.microservice }}
+    cliSources: '${{ parameters.workDir }}/src,${{ parameters.workDir }}/e2e'
+    extraProperties: |
+      sonar.coverage.exclusions=**/*.spec.ts,${{ parameters.workDir }}/e2e/**/*
+      sonar.javascript.lcov.reportPaths=${{ parameters.workDir }}/coverage/${{ parameters.workDir }}/lcov.info
+  displayName: Prepare Analysis for ${{ parameters.microservice }} --> Task
+
+- script: npm ci
+  workingDirectory: ${{ parameters.workDir }}
+  displayName: Install dependencies for ${{ parameters.microservice }} --> Task
+
+- script: npm build --prod
```

Comment:
this isn't right. either do `npx ng build --prod` or `npm run build -- --prod` (or add/modify the package.json scripts to put the `--prod` in there)